doc/source/users/python/index.rst

```diff
@@ -142,6 +142,50 @@ An example where the function returns a ``bharray`` with one element and of type
 
 .. _interop:
 
+
+
+Sliding views between iterations
+~~~~~~~~~~~~~~~~
+
+It can be useful to increase/decrease the beginning of certain array views between iterations of a loop. This can be achieved using ``slide_view()``, which takes a view into an array and a list of tuples. The first argument of a tuple indicates a dimension, while the second dictates the stride the start of a dimension is increased/decreased by for each iteration. At the moment, ``slide_view()`` does not support boundary checks (if the array is underflowed/overflowed, the behaviour is undefined). ``slide_view()`` only supports changes at the end of each iteration and all ``slide_view()`` calls must be placed at the top of the loop body.
+
+The function signature::
+
+  def slide_view(a, dim_stride_tuples):
+    """Creates a dynamic view within a loop, that updates the given dimensions by the given strides at the end of each iteration.
+
+    Parameters
+    ----------
+    a : array view
+        A view into an array
+    dim_stride_tuples: (int, int)[]
+        A list of (dimension, stride) pairs. For each of these pairs, the dimension is updated by the stride in each iteration of a loop.
+
+    Notes
+    -----
+    No boundary checks are performed. If the view overflows the array, the behaviour is undefined.
+    All slide_view() calls must be at the top of the loop body.
+    All views are changed at the end of an iteration and cannot be performed in the middle of a loop body.
+
+An example of using dynamic views could be writing a loop-based faculty function (from 1 to 10). The loop in numpy looks the following::
```

Comment:
That's not `k!` (factorial), it's `k * (k + 1) / 2` (triangular numbers).

Reply:
You are completely right, I have changed the description now.